test/kmerge-test.hs

```diff
@@ -115,6 +124,34 @@ main = do
                     , testCount "loserTreeMerge"  1191 loserTreeMerge    input5
                     , testCount "mutHeapMerge"    1592 mutHeapMerge      input5
                     ]
+                    -- minimal skew for a levelling merge of 1000 elements.
+                    -- with a tree that gives the long input a short path:
+                    -- 1x500 elements with 1 comparison
+                    -- 4x125 elements with 3 comparisons
+                    -- i.e. target is 2000 total comparisons.
+                , testGroup "levelling-min"
+                    [ testCount "sortConcat"      3729 (L.sort . concat) inputLevellingMin
+                    , testCount "listMerge"       2112 listMerge         inputLevellingMin
+                    , testCount "treeMerge"       2730 treeMerge         inputLevellingMin
+                    , testCount "heapMerge"       2655 heapMerge         inputLevellingMin
+                    , testCount "loserTreeMerge"  2235 loserTreeMerge    inputLevellingMin
+                    , testCount "skewedMerge"     1991 skewedMerge       inputLevellingMin
+                    , testCount "mutHeapMerge"    3021 mutHeapMerge      inputLevellingMin
+                    ]
+                    -- maximal skew for a levelling merge of 1000 elements.
+                    -- with a tree that gives the long input a short path:
+                    -- 1x800 elements with 1 comparison
+                    -- 4x 50 elements with 3 comparisons
+                    -- i.e. target is 1400 total comparisons.
+                , testGroup "levelling-max"
+                    [ testCount "sortConcat"      3872 (L.sort . concat) inputLevellingMax
+                    , testCount "listMerge"       1440 listMerge         inputLevellingMax
```

Comment:
List merge performs well here because it merges the first (i.e. long) input only once. If I instead make the last input the largest, it gets very bad.